### DISCUSSION TITLE
FactoryDoctor: what about a config option to handle the output results?

### DISCUSSION THREAD

**Discussion by blocknotes** [test-prof/test-prof#226] [Ideas]
Hey,
nice idea regarding Factory Doctor :)

What about adding a config option (ex. a block option) to be able to handle the results ([probably this part](https://github.com/test-prof/test-prof/blob/v1.0.7/lib/test_prof/factory_doctor/rspec.rb#L76))?
Some use cases are: redirecting the messages to a file, sending them to an external service, expose them in the CI, using Notifications, etc.

**Comment by palkan** [comment]
Hey!
Yeah, that's a good idea, and we have it in our [backlog](https://github.com/orgs/test-prof/projects/2) (see "Reporters API").

Currently, you can specify a custom IO to print logs; for example, to write Test Prof output to a file:

```ruby
TestProf.configure do |config|
  config.output = File.open("path/to/some/file.txt", "w")
  # don't add ASCII color tags
  config.color = false
end
```

**Reply by blocknotes** [reply]
Ah ok, that's nice too.
I think I could make a try with it for the CI 👍
